### DISCUSSION TITLE
Difference between allgather and and allreduce

### DISCUSSION THREAD

**Discussion by base-y** [deepspeedai/DeepSpeed#1910] [Q&A] [answered]
Hi, I am new to deepspeed. I came across the terms `all_gather` and `all_reduce` while syncing the gradients in multi-node/multi-gpu setting. It seems that both terms mean the same thing but not sure. Can I know what is the difference between those terms?

**Comment by tjruwase** [comment, marked as answer]
Please see torch's [all_gather](https://pytorch.org/docs/stable/distributed.html#torch.distributed.all_gather) and [all_reduce](https://pytorch.org/docs/stable/distributed.html#torch.distributed.all_gather) for descriptions.
